### PR TITLE
[WIP] Multi-dimensional voting

### PR DESCRIPTION
**Pull request by michaelkeenan**
This change allows multi-dimensional voting – just Agreement and Overall at first. A post can set its votingSystem from "Classic" to "TwoFactorAgree" to use the new system.

Instead of sending single votes, the client prepares a "ballot" of votes (e.g. an Overall vote and/or an Agreement vote) and sends that. On receiving a ballot, the server cancels all previous votes, and adds the ballot as a new vote. We've kept the original fields of power and voteType, and these still track Overall votes. The new fields voteTypesRecord and powersRecord are objects that contain the Agreement vote (and also the Overall vote, duplicated from the power and voteType fields) and any future vote dimensions.

VoteableTypes (e.g. comments and posts) still have the baseScore and voteCount fields, which still track Overall votes. They also have baseScoresRecord and voteCountsRecord, which mirror voteTypesRecord and powersRecord.

Screen Shot 2021-12-08 at 12 48 19 PM

### CODE REVIEW

File: packages/lesswrong/components/comments/CommentsItem/CommentsItem.tsx [hunk 1]
```diff
@@ -308,7 +314,14 @@ export const CommentsItem = ({ treeOptions, comment, nestingLevel=1, isChild, co
               collection={Comments}
               hideKarma={post?.hideCommentKarma}
             />
-
+            {(votingSystem === 'TwoFactorAgree') && <div className = {classes.agreeButtons}>
```
Comment (jimrandomh) [inline packages/lesswrong/components/comments/CommentsItem/CommentsItem.tsx:317]: This is wrong. All of the vote-buttons on a single comment need to be sharing a single `useVote` call; what you should have done was put the second pair of buttons inside of `SmallSideVote`, not add a second usage of the `SmallSideVote` component. Doing it this way means that the buttons don't have a unified state, which is going to cause problems in client-side updating.

**Comment by jpaddison3** [comment]
[Haven't yet looked at the code.] Can you strongly vote on agreement? And I believe from reading your description that agreement doesn't affect the ordering of the comment, nor the user's karma, is that right?

**Comment by michaelkeenan** [comment]
Sorry, just now saw your comment JP. You can strongly vote on agreement, and agreement currently has no effect on ordering nor on the user's karma.

**Comment by s-cheng** [comment]
Just making sure, this doesn't need a review and can be closed right?

**Pull request closed without merging**